ironrdp/src/rdp/client_info.rs

```diff
@@ -430,52 +430,74 @@ impl PduParsing for TimezoneInfo {
 }
 
 #[derive(Debug, Clone, PartialEq)]
-pub struct SystemTime {
-    pub month: Month,
-    pub day_of_week: DayOfWeek,
-    pub day: DayOfWeekOccurrence,
-    pub hour: u16,
-    pub minute: u16,
-    pub second: u16,
-    pub milliseconds: u16,
+pub enum SystemTime {
```

Comment:
wouldn't it be better to use Option<SystemTime>? it looks a bit odd to have an enum that contains a subenum or whatever this is called, one of which is valid and the other being invalid.